### ISSUE TITLE
Peers keep reconnecting every 30s

### ISSUE THREAD

**Issue by sneaker1** [libp2p/js-libp2p#1598]
- **Version**:
```
    "@chainsafe/libp2p-gossipsub": "^6.2.0",
    "@chainsafe/libp2p-noise": "^11.0.1",
    "@libp2p/mplex": "^7.1.1",
    "@libp2p/peer-id-factory": "^2.0.1",
    "@libp2p/pubsub-peer-discovery": "^8.0.0",
    "@libp2p/tcp": "^6.1.2",
    "@libp2p/websockets": "^5.0.3",
    "libp2p": "^0.42.2"
```

- **Platform**:
Linux 5.15.0-60-generic

- **Subsystem**:
Connection Manager

#### Severity:
Medium

#### Description:
The problem described in #1513 is not solved yet.
With pubsub-peer-discovery the nodes can discover each other, but after ~30s the connection gets disconnected and shortly after reconnected.
In #1513 it was suggested to disable some tcp timeouts with 
```
transports: [tcp({
        outboundSocketInactivityTimeout: 0,
        inboundSocketInactivityTimeout: 0
      })]
```
I tried that, but it is still disconnecting every 30s.
The problem also occurs when using websockets as transport.

#### Steps to reproduce the error:
I updated the example-repository from AlexMesser with the latest versions.
https://github.com/sneaker1/peer-reconnect-problem

Reproduction

```
git clone https://github.com/sneaker1/peer-reconnect-problem
cd peer-reconnect-problem
npm i
npm run relay
npm run alice
npm run bob
```

**Comment by p-shahi** [comment]
Hi @nikumbgo12345 those folks aren't maintaining js-libp2p anymore, if you would like to take this up, please go ahead!

**Comment by nikumbgo12345** [comment]
thanks will do

**Comment by achingbrain** [comment]
The connections that get closed are the circuit relay client connections.  This looks like it's happening because the transport applies the hop timeout to the connection rather than the server only applying it during the initial handshake.

This is with circuit relay v1, the next libp2p release will contain circuit relay v2 which has been completely rewritten, so I'd like to look at this again after the next release.

**Comment by sneaker1** [comment]
Yes, i can confirm this is now working with circuit relay v2 and libp2p version v0.43.x
I have updated my repo with a working example with circuit relay v2 under the branch circuitv2
https://github.com/sneaker1/peer-reconnect-problem/tree/circuitv2

Thanks for your great work.
